e2e/specs/immunizations.spec.ts

```diff
@@ -55,6 +55,80 @@ test('Add an immunization', async ({ page }) => {
     await expect(immunizationType).toContainText(/hepatitis b vaccination/i);
     await expect(vaccinationDate).toContainText(/mar 8, 2024/i);
   });
+
+  await test.step('When I click the expand All rows in the table header with the newly recorded immunization', async () => {
+    await page.getByRole('button', { name: /expand all rows/i }).click();
+  });
+
+  await test.step('Then I should see the immunization saved row in the table', async () => {
+    await expect(page.getByText(/dose number within series/i)).toBeVisible();
+    await expect(page.getByText(/vaccination date/i)).toBeVisible();
+  });
+
+  await test.step('And I click on the `Edit` button', async () => {
+    await page.getByRole('button', { name: /edit/i }).click();
+  });
+
+  await test.step('Then I should see the immunization form launch in the workspace in edit mode`', async () => {
+    await expect(page.getByText(/immunization form/i)).toBeVisible();
+    await expect(page.getByRole('cell', { name: /hepatitis b vaccination/i })).toBeVisible();
+  });
+
+  await test.step('When I change the immunization type to `measles vaccination`', async () => {
+    await page.getByRole('combobox', { name: /immunization/i }).click();
+    await page.getByText(/measles vaccination/i).click();
+  });
+
+  await test.step('And I change `08/03/2024` to `02/01/2025` as the updated vaccination date', async () => {
+    await page.getByLabel(/vaccination date/i).clear();
+    await page.getByLabel(/vaccination date/i).fill('02/01/2025');
+    await page.getByLabel(/vaccination date/i).press('Tab');
+  });
+
+  await test.step('And I change the immunization Dose number within series to `2`', async () => {
+    await page.getByRole('spinbutton', { name: /dose number within series/i }).clear();
+    await page.getByRole('spinbutton', { name: /dose number within series/i }).fill('2');
+  });
+
+  await test.step('And I click on the `Save` button', async () => {
+    await page.getByRole('button', { name: /save/i }).click();
+  });
+
+  await test.step('Then I should see a success notification', async () => {
+    await expect(page.getByText(/vaccination saved successfully/i)).toBeVisible();
+  });
+
+  await test.step('Then I attempt to click the Collapse All Rows button in the table header if visible', async () => {
+    const collapseButton = page.getByRole('button', { name: /collapse all rows/i });
+    if (!(await collapseButton.isVisible())) {
```

Comment:
Do you really need this logic?

Reply:
yes locally this is not checked however on the CI needs to be checked because sometimes after reloading the page the collapse comes expanded instead

Reply:
What do you think of having a simplified code?

```
await test.step('Click Collapse All Rows button if visible', async () => {
  const collapseButton = page.getByRole('button', { name: /collapse all rows/i });
  if (await collapseButton.isVisible()) collapseButton.click();
});
```